app/models/services/tumblr.rb

```diff
@@ -16,12 +16,15 @@ def consumer_secret
   def post(post, url='')
     body = build_tumblr_post(post, url)
     user_info = JSON.parse(client.get("/v2/user/info").body)
-    blogs = user_info["response"]["user"]["blogs"].map { |blog| URI.parse(blog['url']) }
+    blogs = user_info["response"]["user"]["blogs"]
     tumblr_ids = {}
     blogs.each do |blog|
-      resp = client.post("/v2/blog/#{blog.host}/post", body)
-      if resp.code == "201"
-        tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+      if blog["primary"]
+        blogurl = URI.parse(blog["url"])
+        resp = client.post("/v2/blog/#{blogurl.host}/post", body)
+        if resp.code == "201"
+          tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+        end
       end
```

Comment:
The fallback is easy enough using the optional ifnone argument of Enumerable#find :
```
blogs = user_info["response"]["user"]["blogs"]
blog = blogs.find(blogs[0]) {|blog| blog["primary"]}
```
But I'm insufficiently familiar with rspec to be able to write the spec correctly, at least with any reasonable speed